### ISSUE TITLE
Sync downloads all files again

### ISSUE THREAD

**Issue by mhastu** [nextcloud/android#13079]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Sync an online folder to the phone by pressing the three dots and selecting "sync"
2. Add files to the online folder
3. Sync again

### Expected behaviour

Only new or updated files get downloaded.

### Actual behaviour

All files in the folder get downloaded again.

### Android version

12

### Device brand and model

FP4

### Stock or custom OS?

Custom (explain in "additional information")

### Nextcloud android app version

3.29.0

### Nextcloud server version

9.0.2.2

### Using a reverse proxy?

No

### Additional information

Before the sync, a lot of files show the green check icon in the app, after starting the sync, they all change to the syncing icon.

**Comment by Szwendacz99** [comment]
Yeah, sync is broken in general lately, this is not the only problem. At least the FDroid version.

**Comment by paveldvorak5** [comment]
Yes, it stops with an error and don't start again. After manual sync it starts from the beginning.

**Comment by satvik2131** [comment]
can you upload a video , I am not able to reproduce it

**Comment by Szwendacz99** [comment]
> can you upload a video , I am not able to reproduce it

Video should not be needed here, as everything that would be on it is already described, maybe except the weird fact, that for all those files that are being redownloaded, the progress shown in notification seem to be broken - it is not showing progress, but rather the progress bar is almost instantly full, and the percentage goes (also instantly) all the time to the same weird number which is `21474836447%`.

I don't know if that makes difference, but I am using FDroid version of apk.

**Comment by paveldvorak5** [comment]
I noticed that this is logged way too many times, too frequent updates? Once a second is good enough
```
ApplicationPolicy    D  isStatusBarNotificationAllowedAsUser: packageName = com.nextcloud.client,userId = 0
Notification4N_GO  W  PackageUtil.checkIgnoreCase-On-Going,[com.nextcloud.client]

```

The notification says error and I found this in the log:
E  FilesDownloadWorker successfully completed
Worker result SUCCESS for Work
Stopping foreground service
Job didn't exist in JobStore: 4e09136 #u0a531/873 com.nextcloud.client/androidx.work.impl.background.systemjob.SystemJobService

[nextcloud.log](https://github.com/user-attachments/files/15810439/nextcloud.log)

**Comment by paveldvorak5** [comment]
Only 10% was downloaded so why FilesDownloadWorker successfully completed?

**Comment by rhusiev** [comment]
Has there been any progress on this since June? Experiencing a similar issue with 2-way sync.

I have ~1500 files in a directory with sync flag enabled, and with each change all 1500 files sync again

**Comment by gpoole** [comment]
I think this might be the same as #13738?

**Comment by timkrief** [comment]
Hi, I can confirm this is still an issue with 3.30.2 . Also, if all the files in a folders are synced, a synced green tick should also appear on the folder to tell it's entirely synced. Right now, there's no way to tell the status of sync of files inside a folder.

**Comment by timkrief** [comment]
> I think this might be the same as [#13738](https://github.com/nextcloud/android/issues/13738)?

No this is not the same, this issue is about intentionally syncing one folder and all the files in that folder, already synced or not, are being synced again.

Also, I think this issue might be a duplicate of #12641 or closely related.

**Comment by IdeallyGrey** [comment]
Latest release from today ([1.30.4](https://github.com/nextcloud/android/releases/tag/stable-3.30.4)) includes a bug fix, `Optimized to skip syncing already downloaded files.` That might address this issue?

**Comment by joshtrichards** [comment]
Fixed in #13897
